bundles/org.openhab.binding.modbus/README.md

```diff
@@ -985,6 +988,18 @@ end
 
 Please be aware that `REFRESH` commands are "throttled" (to be exact, responses are cached) with `poller` parameter `cacheMillis`.
 
+## Troubleshooting
+
+Modbus, while simple at its heart, potentially is a complicated standard to use because there's a lot of freedom (and bugs) when it comes to implementations.
+There are many device or vendor specific quirks and wrinkles you might stumble across. Here's some:
+
+* a device may have more than one interface with a port open (502 or other) that it responds on, but may have no connection to the real bus hardware.
+Check ALL interfaces including (and preferring) any serial connection. Usually either that or the Ethernet IP will do.
+
+* some devices do not allow to query a range of registers that is too large or spans reserved registers.
+They may respond with an error or no error but invalid register data.
```

Comment:
Can we make the troubleshooting more actionable?

For example, trying first to read large spans (more efficient especially with tens of registers with short polling), containing even undefined/undocumented data regions.

And if that does not work, trying to read the data in pieces that contains only documented/valid data regions.


We have seen all sorts of behavior 
- ok to query undefined registers, in the end you would only use the documented registers. 
- slave not responding at all if the data region contains undefined data. Will produce timeout error in binding 
- slave responding with explicit error, telling that addresses are invalid. Thing will go offline and description will tell the reason

The weirdest is where you need to query 4 registers, even if you are only needing the first two. This is an issue with SMA devices https://github.com/openhab/openhab1-addons/issues/4931#issue-197499556. Comes up every now and then as people stumble on it.

Reply:
Perhaps the advice would be to review if any of the device specific modbus bindings are applicable for your device, e.g. Sunspec modbus binding. Many quirks are taken cared for and it's much easier for the user than the generic modbus binding.

I think the list is in the beginning of the document.

What do you think?

Reply:
I've put up a couple of changes and hope they match your comments.

Regarding the last one, this doesn't belong into troubleshooting IMHO so I didn't do anything about it.
It should be more prominently emphasized than now at the beginning of the document though.